ShenkinStore/Controllers/ShoppingCartController.cs

```diff
@@ -0,0 +1,106 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using ShenkinStore.Models;
+using ShenkinStore.ViewModels;
+
+namespace ShenkinStore.Controllers
+{
+    public class ShoppingCartController : Controller
```

Comment:
Make sure you checked the scenario of switching accounts and going back, and logout and login again (cart should be saved).

Reply:
already checked, the carts are saved